src/gui/src/dbDescriptors.cpp

```diff
@@ -1639,4 +1648,139 @@ bool DbTechLayerDescriptor::getAllObjects(SelectionSet& objects) const
   return true;
 }
 
+//////////////////////////////////////////////////
+
+static const char*
+getAccessTypeName(odb::dbAccessPoint::AccessType type)
+{
+  switch(type) {
+  case odb::dbAccessPoint::OnGrid:
+    return "OnGrid";
+  case odb::dbAccessPoint::HalfGrid:
+    return "HalfGrid";
+  case odb::dbAccessPoint::Center:
+    return "Center";
+  case odb::dbAccessPoint::EncOpt:
+    return "EncOpt";
+  case odb::dbAccessPoint::NearbyGrid:
+    return "NearbyGrid";
+  }
+  return "Unknown";
+}
+
+DbItermAccessPointDescriptor::DbItermAccessPointDescriptor(odb::dbDatabase* db) :
+    db_(db)
+{
+}
+
+std::string DbItermAccessPointDescriptor::getName(std::any object) const
+{
+  auto iterm_ap = std::any_cast<DbItermAccessPoint>(object);
+  auto ap = iterm_ap.ap;
+  std::string name(getAccessTypeName(ap->getLowType()));
+  name += std::string("/") + getAccessTypeName(ap->getHighType());
+  return name;
+}
+
+std::string DbItermAccessPointDescriptor::getTypeName() const
+{
+  return "Access Point";
+}
+
+bool DbItermAccessPointDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  auto iterm_ap = std::any_cast<DbItermAccessPoint>(object);
+  odb::Point pt = iterm_ap.ap->getPoint();
+  odb::dbTransform xform;
+  int x, y;
+  iterm_ap.iterm->getInst()->getLocation(x, y);
+  xform.setOffset({x, y});
```

Comment:
What happens if the instance is rotated? Would it be better to get the instance transform instead of just the X/Y?

Reply:
Different orientations get different sets of APs so the APs themselves are represented without orientation as they only apply to one possible orientation.